docker/docker_image_dest.go

```diff
@@ -326,7 +326,7 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 	if err != nil {
 		return false, private.ReusedBlob{}, err
 	}
-	if haveBlob {
+	if haveBlob && (options.RequiredCompression == nil || (options.RequiredCompression.Name() == info.CompressionAlgorithm.Name())) {
```

Comment:
Moving this before `tryReusingExactBlob` would need me to add a condition like this
```go
       if !impl.BlobMatchesRequiredCompression(options, info.CompressionAlgorithm) {
                return false, private.ReusedBlob{}, nil
        }
```

but in this case since substitute is allowed I want other candidates to be checked further below in the code, so it seemed appropriate to check with when `exact` blob ( i.e `haveBlob`) is found and return `true` for that blob if it matches the condition as compared to returning `false` early.

Reply:
Would
```go
if matchesCompression {
   if tryReusingExactBlob() succeeded {
     return success
  }
}
… Cache.CacheLocations2 …
```
not work?

Reply:
yes that would work